### DISCUSSION TITLE
refreshInterval as function

### DISCUSSION THREAD

**Discussion by correttojs** [vercel/swr#1630] [Ideas]
currently the `refreshInterval` option can only be a number. In some cases this number should be dynamic, dependent on the updated fetched data.
For example, the fetcher returns a list of items and a value corresponding to the next expected update.

**Comment by nilskaspersson** [comment]
I would appreciate this functionality as well, and in particular would I like to reuse the exponential back-off behavior used in `onErrorRetry`. My use-case is that I occasionally have eventually consistent data, but no way of knowing when. Usually it's within seconds, but sometimes hours—but I only need it conditionally. 

I'm happy to draft a PR, but I would like to see some discussion on it first, to see if it's even wanted in the first place.

**Comment by shuding** [comment]
This is now landed in 1.1.1 ([release](https://github.com/vercel/swr/releases/tag/1.1.1)). 🎉
